### PR TITLE
Added commands for listing activity windows.

### PR DESCRIPTION
**Pull request by akhadem**
Listing activity windows is a new functionality as part of the new rolled monitoring service on ADF. this PR adds commands to use that. Help file update with the right information too.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@akhadem__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (akhadem). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/ActivityWindowContextBaseCmdlet.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+namespace Microsoft.Azure.Commands.DataFactories
+{
+    public abstract class ActivityWindowContextBaseCmdlet : DataFactoryBaseCmdlet
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/ActivityWindowContextBaseCmdlet.cs]: ?? WHy an abstract class with no members?

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The state of the activity window. The possible states includes: None, Waiting, InProgress, Ready, Failed, and Skipped")]
+        [ValidateNotNullOrEmpty]
+        public string WindowState { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The substate of the activity window. The possible substates includes: Canceled, timedOut, Validating," +
+                                                                                                          " ScheduledTime, DatasetDependencies, ComputeResources, ConcurrencyLimit, ActivityResume, Retry, Validation, ValidationRetry.")]
+        [ValidateNotNullOrEmpty]
+        public string WindowSubstate { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The activity window filter based on Azure Search filter grammar. For more information see https://msdn.microsoft.com/en-us/library/azure/dn798921.aspx")]
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs]: How can you have so many parameters with Position 4?  Are there meant to be multiple parameter sets?

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        [Parameter(Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The dataset name.")]
+        [ValidateNotNullOrEmpty]
+        public string DatasetName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The pipeline name.")]
+        [ValidateNotNullOrEmpty]
+        public string PipelineName { get; set; }
+
+        [Parameter(Position = 3, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The pipeline activity name.")]
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs]: Please fix line breaks at 80 cols.

File: src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.ActivityWindows.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+using Microsoft.Azure.Management.DataFactories.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Microsoft.Azure.Commands.DataFactories
+{
+    public partial class DataFactoryClient
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.ActivityWindows.cs]: Fix line breaks

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/ActivityWindowContextBaseCmdlet.cs (hunk 1 above)
Comment (akhadem) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/ActivityWindowContextBaseCmdlet.cs]: removed.

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs (hunk 2 above)
Comment (akhadem) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs]: Yes, they are all multiple optional parameters that user can leverage or not. There is no priority or preference

File: src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs (hunk 3 above)
Comment (akhadem) [inline src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs]: Tried to do in as many places as possible

**Pull request merged**